bin/test

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+main() {
+    # parse the arguments into variables
+    declare build_dir="${1:-}"
```

Comment:
Why `declare`? You meant `local`?

Reply:
`bin/test` doesn't receive arguments, so this whole bit is redundant anyway.

Reply:
Looks like it does, based on the [nodejs buildpack](https://github.com/heroku/heroku-buildpack-nodejs/blob/master/bin/test).

Reply:
I was responding to this bin receiving arguments, not the declare vs local vs not assigning.

Reply:
I referred to the nodejs buildpack, like @josegonzalez wrote. The idea behind this declaration is that everyone who looks at the code exactly knows what the first argument of this function does. It's just for better readability.

`declare`is used in here, as `local`shifts the arguments. E.g.

```
main() {
    local arg1=$1 # would be $1 with declare
    local arg2=$1 # would be $2 with declare
}
```

Reply:
@dj95 for consistency's sake, lets just move to not having a function? I don't think thats the way any of the buildpacks are structured for bins (unfortunately) so introducing a new pattern isn't great :(

We can just add a comment at the top with a list of args etc. this thing is expecting.

Reply:
I removed the function, as you can see in the new commits, and moved the declare statement to the top in order to keep the readability :)